src/PageLayout/interaction.stories.tsx

```diff
@@ -0,0 +1,429 @@
+        value: ['none', 'line']
+      },
+      defaultValue: 'none',
+      control: {
+        type: 'radio'
+      },
+      table: {
+        category: 'Footer props'
+      }
+    },
+    'Footer.padding': {
+      type: {
+        name: 'enum',
+        value: ['none', 'condensed', 'normal']
+      },
+      defaultValue: 'none',
+      control: {type: 'radio'},
+      table: {category: 'Footer props'}
+    },
+    'Footer.hidden.regular': {
+      type: 'boolean',
+      defaultValue: false,
+      table: {category: 'Footer props'}
+    },
+    'Footer.hidden.narrow': {
+      type: 'boolean',
+      defaultValue: false,
+      table: {category: 'Footer props'}
+    },
+    'Footer.hidden.wide': {
+      type: 'boolean',
+      defaultValue: false,
+      table: {category: 'Footer props'}
+    }
+  }
+}
+
+StickyPane.argTypes = {
+  sticky: {
+    type: 'boolean',
+    defaultValue: true
+  },
+  numParagraphsInPane: {
+    type: 'number',
+    defaultValue: 3
+  },
+  numParagraphsInContent: {
+    type: 'number',
+    defaultValue: 30
+  }
+}
+
+const isInViewPort = (boundingRect: {top: number; left: number; right: number; bottom: number}) => {
+  return (
+    boundingRect.top >= 0 &&
+    boundingRect.left >= 0 &&
+    boundingRect.right <= document.documentElement.clientWidth &&
+    boundingRect.bottom <= document.documentElement.clientHeight
+  )
+}
+
+StickyPane.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
+  const canvas = within(canvasElement)
+  const content3 = await canvas.getByTestId('content3')
+  content3.scrollIntoView()
+  const paragraph0 = await canvas.getByTestId('paragraph0')
+  const paragraphRect = paragraph0.getBoundingClientRect()
+  expect(isInViewPort(paragraphRect)).toBe(true)
+}
+
+const NonStickyPane = StickyPane.bind({})
+NonStickyPane.argTypes = {
+  sticky: {
+    type: 'boolean',
+    defaultValue: false
+  },
+  numParagraphsInPane: {
+    type: 'number',
+    defaultValue: 6
+  },
+  numParagraphsInContent: {
+    type: 'number',
+    defaultValue: 30
+  }
+}
+
+NonStickyPane.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
+  const canvas = within(canvasElement)
+  const content3 = await canvas.getByTestId('content3')
+  content3.scrollIntoView()
+  const paragraph0 = await canvas.getByTestId('paragraph0')
+  const paragraphRect = paragraph0.getBoundingClientRect()
+  expect(isInViewPort(paragraphRect)).toBe(false)
+}
+
+CustomStickyHeader.argTypes = {
+  sticky: {
+    type: 'boolean',
+    defaultValue: true
+  },
+  stickyTop: {
+    type: 'string',
+    defaultValue: '8rem'
+  },
+  numParagraphsInPane: {
+    type: 'number',
+    defaultValue: 10
+  },
+  numParagraphsInContent: {
+    type: 'number',
+    defaultValue: 30
+  }
+}
+
+CustomStickyHeader.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
+  const canvas = within(canvasElement)
+  const contentToScroll = await canvas.getByTestId('content3')
+  contentToScroll.scrollIntoView()
```

Comment:
Do you need to scroll twice? The `fireEvent.scroll` below would be ideal. I would fix my tests to work on that if it works for you.

Reply:
No it doesn't work :( This is what I mean [here ](https://github.com/primer/react/pull/2241#discussion_r951030514) Maybe I am missing something? 

Because the fireEvent.scroll didn't work I included contentToScroll. I'll give another go today and let you know if I make it work.

Reply:
Yeah! I had the same experience. It would have been so much better if fireEvent scroll worked in this case.Maybe we aren't using the scrollcontainer effectively?

Reply:
I tried it again but no luck 😢 I agree, I think it is about the scroll container. I tried the outer box, the page layout content as scroll containers but none of them are scrollable. (el.scrollHeight-el.clientHight is 0) and I tried giving auto overflow but that doesn't seem to work either. I'll remove this fireEvent for now and maybe down the track, it will be possible to do with user events from SB.